src/main/java/net/mcreator/ui/views/editor/image/canvas/Canvas.java

```diff
@@ -30,69 +29,91 @@
 import java.util.UUID;
 
 public class Canvas extends ArrayListListModel<Layer> {
+
+	//** Transient references and fields **//
+	// ImageMakerView reference (needs to be set right after creation)
+	private transient ImageMakerView imageMakerView;
+
+	// Preview fields
+	private transient boolean drawPreview = false;
+	private transient Shape shape = Shape.SQUARE;
+	private transient int size = 1;
+
+	// Preview image (if custom image is provided for preview)
+	private transient Image previewImage;
+	private transient boolean drawCustomPreview;
+
+	// Preview changed event
+	private transient MouseEvent previewEvent;
+
+	// Pasted layer reference
+	private transient Layer floatingLayer = null;
+
+
+	//** Saved image properties **//
+	// Canvas size
 	private int width;
 	private int height;
-	private final LayerPanel layerPanel;
-	private final VersionManager versionManager;
+
+	// Selection object
 	private final Selection selection;
-	private CanvasRenderer canvasRenderer;
-	private boolean drawPreview;
-	private MouseEvent previewEvent;
-	private Shape shape = Shape.SQUARE;
-	private int size;
 
-	private Image previewImage;
-	private boolean drawCustomPreview;
-	private Layer floatingLayer;
 
-	public Canvas(int width, int height, LayerPanel layerPanel, VersionManager versionManager) {
+	public Canvas(int width, int height) {
 		this.width = width;
 		this.height = height;
-		this.layerPanel = layerPanel;
-		this.versionManager = versionManager;
 		this.selection = new Selection(this);
-		layerPanel.setCanvas(this);
-		layerPanel.updateSelection();
+	}
+
+	public void initReferences(ImageMakerView imageMakerView) {

```

Comment:
Make this private and call from constructor where constructor also takes ImageMakerView imageMakerView